File: src/services/mcp_service.ts
import { invoke } from '@tauri-apps/api/core';
import type {
  McpService,
  McpServiceDiscovery,
} from '@/types/mcp-service';

export async function listMcpServices(): Promise<McpService[]> {
  return invoke<McpService[]>('list_mcp_services');
}

export async function upsertMcpService(service: McpService): Promise<string> {
  return invoke<string>('upsert_mcp_service', { service });
}

export async function deleteMcpService(serviceId: string): Promise<string> {
  return invoke<string>('delete_mcp_service', { serviceId });
}

export async function discoverMcpService(service: McpService): Promise<McpServiceDiscovery> {
  return invoke<McpServiceDiscovery>('discover_mcp_service', { service });
}

File: src-tauri/src/commands/mcp_service.rs
use crate::db::local_store::LocalJsonStore;
use crate::db::repositories::mcp_service::McpServiceRepository;
use crate::models::mcp_service::{McpService, McpServiceDiscovery};
use crate::services::mcp_service::discover_service;

#[tauri::command]
pub fn list_mcp_services(
    store: tauri::State<'_, LocalJsonStore>,
) -> Result<Vec<McpService>, String> {
    let repo = McpServiceRepository::new(&store);
    repo.list()
}

#[tauri::command]
pub fn upsert_mcp_service(
    store: tauri::State<'_, LocalJsonStore>,
    service: McpService,
) -> Result<String, String> {
    let repo = McpServiceRepository::new(&store);
    repo.upsert(&service)
}

#[tauri::command]
pub fn delete_mcp_service(
    store: tauri::State<'_, LocalJsonStore>,
    service_id: String,
) -> Result<String, String> {
    let repo = McpServiceRepository::new(&store);
    repo.delete(&service_id)
}

#[tauri::command]
pub async fn discover_mcp_service(
    service: McpService,
) -> Result<McpServiceDiscovery, String> {
    discover_service(&service).await
}
